lib/client.js

```diff
@@ -1,11 +1,41 @@
 var PROMISE_TIMEOUT = 5000, // 5 seconds
+    PROMISE_DONT_TIMEOUT = ['instances.create'],
     ZAF_EVENT       = /^zaf\./,
     version         = require('version'),
+    Utils           = require('utils'),
     Promise         = window.Promise || require('../vendor/native-promise-only'),
     when            = require('utils').when,
     pendingPromises = {},
     ids             = {};
 
+// @internal
+// #timeoutReject(name, params)
+//
+// Reject a request if required, given the rejection function, name (get, set or invoke)
+// and arguments to that request. Falls back to 5000 second timeout with few exceptions.
+//
+function timeoutReject(reject, name, paramsArray) {
```

Comment:
#nit `timeoutReject` bothers me slightly, like it takes a little more effort to understand what it's actually doing.